kolibri/core/content/api.py

```diff
@@ -1321,6 +1326,72 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentDownloadRequestViewset(
+    ReadOnlyValuesViewset, CreateModelMixin, DestroyModelMixin
+):
+    serializer_class = serializers.ContentDownloadRequestSeralizer
+
+    pagination_class = OptionalPageNumberPagination
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+        "facility",
+        "source_id",
+    )
+
+    def get_queryset(self):
+        return ContentDownloadRequest.objects.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return queryset.annotate(
+            has_removal=Exists(
+                ContentRemovalRequest.objects.filter(
+                    source_model=OuterRef("source_model"),
+                    source_id=OuterRef("source_id"),
+                    contentnode_id=OuterRef("contentnode_id"),
+                    requested_at__gte=OuterRef("requested_at"),
+                )
+            )
+        ).filter(has_removal=False)
+
+    def delete(self, request, *args, **kwargs):
```

Comment:
I wasn't sure, so I got into a little bit of a rabbit hole about whether DELETE requests can actually have a request body. It seems that they can, but it is strongly discouraged. So let's not take this before the web standards panel any time soon - but doing this via a URL param is just as weird, as its specific per user.